File: heritage_simple/cpp/redefinition/Chien.cpp
#include "Chien.h"

Chien::Chien():Animal(){};
Chien::Chien(string nom, int nombreDePatte):Animal(nom,nombreDePatte){};
Chien::Chien(Chien const& chien):Animal(chien){};

//redéfinition de la méthode crier

inline void Chien::crier() const{
    cout<<"Whouaf whouaf !!!"<<endl;
}

Chien::~Chien(){};

File: heritage_simple/java/redefinition/Chien.java
public class Chien extends Animal{
    public Chien(){
        super("Chien",4);
    }

    public Chien(String nom, int nombreDePattes){
        super(nom, nombreDePattes);
    }

    public Chien(Chien chien){
        super(chien);
    }

    //pas besoin de getter et de setter car deja défini dans la classe Mere Animal

    //redéfinition de la méthode crier
    public void crier(){
        System.out.println("Whouaf whouaf !!!");
    }
}